### ISSUE TITLE
Add search functionality to Competition Center

### ISSUE THREAD

**Issue by Solohan50** [emperorshammer/issues-incoming-queue#118]
Please add the ability to search the Competition Center listings, preferably by the event name and the submitter's name (the latter is the least important of the two).  To find an old competition currently, you have to just keep clicking through pages and reading through the entire list until you find the competition in question.

This feature is likely only necessary in the Past Competitions section, though it wouldn't hurt to have it everywhere.

**Comment by turtlexiv** [comment]
Good idea; we have a larger competitions module improvement planned, but we might be able to work this in before that.

emperorshammer/tiecorps#44
